brooklyn-tosca-transformer/src/test/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaPlanToSpecTransformerTest.java

```diff
@@ -70,21 +76,35 @@ public void testDslInChatApplication() {
         assertEquals(app.getChildren().size(), 2);
 
         EntitySpec<TomcatServer> tomcatServer = (EntitySpec<TomcatServer>) app.getChildren().get(0);
-        assertEquals(tomcatServer.getConfig().size(), 6);
         assertNotNull(app.getChildren().get(0).getConfig().get(TomcatServer.JAVA_SYSPROPS));
 
         Map javaSysProps = (Map) app.getChildren().get(0).getConfig().get(TomcatServer.JAVA_SYSPROPS);
 
         assertEquals(javaSysProps.size(), 1);
         assertTrue(javaSysProps.get("brooklyn.example.db.url") instanceof BrooklynDslDeferredSupplier);
-        assertEquals(javaSysProps.get("brooklyn.example.db.url").toString(),
-                "$brooklyn:formatString(\"jdbc:%s%s?user=%s\\\\&password=%s\"," +
-                        "$brooklyn:entity(\"mysql_server\").attributeWhenReady(\"datastore.url\")," +
-                        "visitors," +
-                        "brooklyn," +
-                        "br00k11n)");
+        assertEquals(javaSysProps.get("brooklyn.example.db.url").toString(), DATABASE_DEPENDENCY_INJECTION);
 
         assertTrue(tomcatServer.getLocations().get(0) instanceof LocalhostMachineProvisioningLocation);
     }
 
+    @Test
+    public void testRelation(){
+        String templateUrl = getClasspathUrlForResource("templates/relationship.yaml");
+
+        EntitySpec<? extends Application> app = transformer.createApplicationSpec(
+                new ResourceUtils(mgmt).getResourceAsString(templateUrl));
+
+        assertNotNull(app);
+
+        assertEquals(app.getChildren().size(), 2);
+
+        EntitySpec<TomcatServer> tomcatServer = (EntitySpec<TomcatServer>) app.getChildren().get(0);
+        assertNotNull(tomcatServer.getConfig().get(TomcatServer.JAVA_SYSPROPS));
```

Comment:
This assertion fails. If app has two children how do you know that `app.getChildren().get(0)` will give you the spec for the Tomcat server?